cmake/TinyCNNConfig.cmake.in

```diff
@@ -0,0 +1,136 @@
+# of the one placed in 'install' directory.
+if (FALSE)
+# Get the (current, i.e. installed) directory containing this file.
+get_filename_component(TINY_CURRENT_CONFIG_INSTALL_DIR
+    "${CMAKE_CURRENT_LIST_FILE}" PATH)
+
+# Record the state of the CMake module path when this script was called so
+# that we can ensure that we leave it in the same state on exit as it was
+# on entry, but modify it locally.
+set(CALLERS_CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH})
+
+# Reset CMake module path to the installation directory of this script,
+# thus we will use the FindPackage() scripts shipped with TinyCNN to find
+# Tiny's dependencies, even if the user has equivalently named FindPackage()
+# scripts in their project.
+set(CMAKE_MODULE_PATH ${TINY_CURRENT_CONFIG_INSTALL_DIR})
+
+# Build the absolute root install directory as a relative path (determined when
+# TinyCNN was configured & built) from the current install directory for this
+# this file.  This allows for the install tree to be relocated, after TinyCNN was
+# built, outside of CMake.
+get_filename_component(CURRENT_ROOT_INSTALL_DIR
+    ${TINY_CURRENT_CONFIG_INSTALL_DIR}/@INSTALL_ROOT_REL_CONFIG_INSTALL_DIR@
+    ABSOLUTE)
+if(NOT EXISTS ${CURRENT_ROOT_INSTALL_DIR})
+    tiny_report_not_found(
+      "TinyCNN install root: ${CURRENT_ROOT_INSTALL_DIR}, "
+      "determined from relative path from TinyCNNConfig.cmake install location: "
+      "${TINY_CURRENT_CONFIG_DIR}, does not exist. Either the install "
+      "directory was deleted, or the install tree was only partially relocated "
+      "outside of CMake after Ceres was built.")
+endif(NOT EXISTS ${CURRENT_ROOT_INSTALL_DIR})
+# Set the include directories for TinyCNN.
+set(TINYCNN_INCLUDE_DIR ${CURRENT_ROOT_INSTALL_DIR}/include)
+if(NOT EXISTS ${TINYCNN_INCLUDE_DIR}/tiny_cnn/tiny_cnn.h)
+     tiny_report_not_found(
+      "TinyCNN install root: ${CURRENT_ROOT_INSTALL_DIR}, "
+      "determined from relative path from TinyCNNConfig.cmake install location: "
+      "${TINY_CURRENT_CONFIG_DIR}, does not contain TinyCNN headers. "
+      "Either the install directory was deleted, or the install tree was only "
+      "partially relocated outside of CMake after TinyCNN was built.")
+endif(NOT EXISTS ${TINYCNN_INCLUDE_DIR}/tiny_cnn/tiny_cnn.h)
+list(APPEND TINYCNN_INCLUDE_DIRS ${TINYCNN_INCLUDE_DIR})
+endif()
+
+# Set the version.
+set(TINYCNN_VERSION @TINYCNN_VERSION@)
+
+# TODO: @tiny_cnn_hdrs@ are not the headers placed
+# inside 'install/include/tiny_cnn'
+
+# Set the headers.
+set(TINYCNN_INCLUDE_DIR @tiny_cnn_hdrs@)
+
+# Set the libraries.
+set(TINYCNN_LIBRARIES "")
+
+# TODO: this checking of required dependencies almost the same as
+# in root CMake. Is there any clean design pattern for this?
+
+# TBB.
+# Append the locations of TBB when TinyCNN was built to the search path hints.
+# Search quietly s/t we control the timing of the error message if not found.
+if(@USE_TBB@)
+	find_package(TBB QUIET REQUIRED)
```

Comment:
right! Besides, is there any good practice to not duplicate the dependencies checking code when we export the package?